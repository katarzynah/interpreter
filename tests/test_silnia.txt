program f;

function silnia (a : int) : int ;
begin
    if a <=1 then silnia:=1
    else silnia := a * silnia(a-1) endif;
end;

begin

print (silnia (6));

end .