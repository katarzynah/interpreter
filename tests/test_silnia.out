[(fromList [],fromList [(Ident "silnia",ProcDecFun (FunHead (Ident "silnia") (Args (ArgList (ArgLabel (IdLEnd (Ident "a")) TypeSpecInt) ArgListEmpty)) TypeSpecInt) (Dec VarDecEmpty ProcDecEmpty) (CompStmnt (StmntList (SIf (IfStmntWithElse (ExpLessOrEqual (SimpleExpTerm (TermFactor (FactorIdent (Ident "a")))) (SimpleExpTerm (TermFactor (FactorConstant (ConstInt 1))))) (SAss (AssStmnt (Ident "silnia") (ExpSimple (SimpleExpTerm (TermFactor (FactorConstant (ConstInt 1))))))) (SAss (AssStmnt (Ident "silnia") (ExpSimple (SimpleExpTerm (TermMultiply (TermFactor (FactorIdent (Ident "a"))) (FactorFunctionCall (FunsCall (Ident "silnia") (Act (ExpList (ExpSimple (SimpleExpSubst (SimpleExpTerm (TermFactor (FactorIdent (Ident "a")))) (TermFactor (FactorConstant (ConstInt 1))))) ExpListEmpty))))))))))) StmntListEmpty)))])]
720
